dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java

```diff
@@ -243,83 +256,216 @@ public static void unzipFile(InputStream zipFile, String outputDir) throws Excep
         }
     }
 
+    /**
+     * Resolves the integrity data path based on provided endpoint id.
+     *
+     * @param endpointId endpoint if
+     * @return integrity data path as a string
+     */
+    public static String getIntegrityDataPath(final String endpointId) {
+        return ConfigUtils.getIntegrityPath() + File.separator + endpointId;
+    }
+
+    /**
+     * Generated the integrity data file path based on provided endpointId and the file name.
+     * File names to be used: DataToCheck.zip, DataToFix.zip and DataStatus.properties.
+     *
+     * @param endpointId endpoint gid
+     * @param dataFile data filename
+     * @return path to filename
+     */
+    public static String getIntegrityDataFilePath(final String endpointId, final String dataFile) {
+        return getIntegrityDataPath(endpointId) + File.separator + dataFile;
+    }
+
+    /**
+     * Tells whether a integrity file exists based on provided endpointId and the file name.
+     *
+     * @param endpointId endpoint id
+     * @param dataFile data filename
+     * @return path to filename
+     */
+    public static boolean doesIntegrityDataFileExist(final String endpointId, final String dataFile) {
+        return new File(getIntegrityDataFilePath(endpointId, dataFile)).exists();
+    }
+
+    /**
+     * Gets a {@link IntegrityDataExecutionMetadata} instance based on provided endpoint id which contains metadata of the
+     * integrity data generation execution.
+     *
+     * @param endpointId endpoint id
+     * @return Optional wrapping the integrity generation  metadata
+     */
+    public static Optional<IntegrityDataExecutionMetadata> getIntegrityMetadata(final String endpointId) {
+        final File statusFile = new File(getIntegrityDataFilePath(endpointId, INTEGRITY_DATA_STATUS_FILENAME));
+        if (!statusFile.exists()) {
+            return Optional.empty();
+        }
+
+        final Properties statusData = new Properties();
+        try {
+            statusData.load(new FileInputStream(statusFile));
```

Comment:
Codacy found an issue: [Avoid instantiating FileInputStream, FileOutputStream, FileReader, or FileWriter](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)

Reply:
should use Files.newInputStream